src/hashmap.c

```diff
@@ -1,51 +1,36 @@
-#define BUCKETS 1000000
-#define PRIME_SEED_1 17
-#define PRIME_SEED_2 97
-#define KEY_CHARS 128
-#define VALUE_CHARS 128
-#define NODE_SIZE sizeof(Node)
-
+#include "hashmap.h"
```

Comment:
Normally you would have the .c file #include the .h file, not the other way around. 

This allows you to compile the .c file separately from any code that includes the .h file, meaning you dont have to recompile the whole library every time you make a change to some code that uses it.

It also means the library becomes its own "compilation unit" meaning it can be built in parallel with other code.

Finally if you want to keep your implementation secret (for business reasons or smth) then you can send just the header file and an object file of the .c file. The header file tells the user how to use the library and the object file can be linked in with the final executable.